sdk/communication/azure-communication-common/src/main/java/com/azure/android/communication/common/AutoRefreshUserCredential.java

```diff
@@ -125,16 +112,28 @@ private synchronized void scheduleTokenFutureUpdate(CommunicationAccessToken acc
         }
 
         long delayMs = 0;
-        if (accessToken != null) {
-            long refreshEpochSecond = accessToken.getExpiresAt().toEpochSecond() - PROACTIVE_REFRESH_BUFFER_SECS;
-            long currentEpochSecond = System.currentTimeMillis() / 1000;
-            delayMs = Math.max((refreshEpochSecond - currentEpochSecond) * 1000, 0);
+        if (accessToken != null && !accessToken.isExpired()) {
+            long currentEpochMs = System.currentTimeMillis();
+            long tokenTtlMs = accessToken.getExpiresAt().toInstant().toEpochMilli() - currentEpochMs;
+            long nextFetchTimeMs = isTokenExpiringSoon(accessToken, PROACTIVE_REFRESH_BUFFER_SECS)
+                ? tokenTtlMs / REFRESH_AFTER_TTL_DIVIDER
+                : tokenTtlMs - TimeUnit.MILLISECONDS.convert(PROACTIVE_REFRESH_BUFFER_SECS, TimeUnit.SECONDS);
+            delayMs = Math.max(nextFetchTimeMs, 0);
         }
 
         this.tokenFutureUpdater = CompletableFuture.runAsync(this::updateTokenFuture,
             CompletableFuture.delayedExecutor(delayMs, TimeUnit.MILLISECONDS));
     }
 
+    private boolean isTokenExpiringSoon(CommunicationAccessToken accessToken, int refreshBufferSecs) {
+        if (accessToken == null) {
```

Comment:
yes, you are right, but I think it's better to leave it as it's, as it can provide a future proof check if it can be reused in the future if there will be a possibility to pass null objects. any thoughts on this?

Reply:
I still believe we should not add checks that is not going to be executed at all since it might create confusion for others who are going over this logic. If needed in future, the developer can still add the null check and test it via test cases.

But I'm fine with both the approaches.